modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -1167,6 +1200,256 @@ SUBC_SIMD(float, float)
+                                                  uchar* outx, const v_float32& sc,
+                                                  const v_float32& scale, const int nlanes)
+{
+    v_float32 a1 = vg_load_f32(inx);
+    v_float32 a2 = vg_load_f32(&inx[nlanes/4]);
+    v_float32 a3 = vg_load_f32(&inx[nlanes/2]);
+    v_float32 a4 = vg_load_f32(&inx[3 * nlanes/4]);
+
+    vx_store(outx, v_pack_u(v_pack(v_round(scale * a1* sc),
+                                   v_round(scale * a2* sc)),
+                            v_pack(v_round(scale * a3* sc),
+                                   v_round(scale * a4* sc))));
+}
+
+//-------------------------------------------------------------------------------------------------
+
+template<typename SRC>
+CV_ALWAYS_INLINE void mulc_scale_simd_common_impl(const SRC* inx,
+                                                  float* outx, const v_float32& sc,
+                                                  const v_float32& scale, const int)
+{
+    v_float32 a1 = vg_load_f32(inx);
+    vx_store(outx, scale * a1* sc);
+}
+
+//-------------------------------------------------------------------------------------------------
+
+template<typename SRC, typename DST>
+CV_ALWAYS_INLINE int mulc_scale_simd_common(const SRC in[],
+                                            const float scalar[], DST out[],
+                                            const int length, const float _scale)
+{
+    constexpr int nlanes = vector_type_of_t<DST>::nlanes;
+
+    if (length < nlanes)
+        return 0;
+
+    v_float32 _scalar = vx_load(scalar);
+    v_float32 scale = vx_setall_f32(_scale);
+
+    int x = 0;
+    for (;;)
+    {
+        for (; x <= length - nlanes; x += nlanes)
+        {
+            mulc_scale_simd_common_impl(&in[x], &out[x], _scalar, scale, nlanes);
+        }
+
+        if (x < length)
+        {
+            x = length - nlanes;
+            continue;  // process unaligned tail
+        }
+        break;
+    }
+    return x;
+}
+
+#define MULC_SIMD(SRC, DST)                                                    \
+int mulc_simd(const SRC in[], const float scalar[], DST out[],                 \
+              const int length, const int chan, const float scale)             \
+{                                                                              \
+    mul_tag op_t;                                                              \
+    switch (chan)                                                              \
+    {                                                                          \
+    case 1:                                                                    \
+    case 2:                                                                    \
+    case 4:                                                                    \
+    {                                                                          \
+        if (std::fabs(scale - 1.0f) <= FLT_EPSILON)                            \
```

Comment:
could you put comment please about what happened here?
if scale ~ 1, then we use scalar version?

Reply:
If scale = 1.0, we go to the branch without scaling, i.e. a*scalar only.